docs/security-privacy-self-review.md

```diff
@@ -1,102 +1,78 @@
+## Responses to the [Self-Review Questionnaire: Security and Privacy](https://w3ctag.github.io/security-questionnaire/) for the [Direct Sockets API](https://github.com/WICG/direct-sockets/)
 
 
-Responses to the [Self-Review Questionnaire: Security and Privacy](https://w3ctag.github.io/security-questionnaire/) for the [Direct Sockets API](https://github.com/WICG/direct-sockets/)
+1. **What information might this feature expose to Web sites or other parties, and for what purposes is that exposure necessary?**
 
+    When the client establishes a socket connection with a remote endpoint, and the connection succeeds, that system's IP address and port, and any information it sends to the remote endpoint, becomes available to the client, and vice versa.
 
+2. **Do features in your specification expose the minimum amount of information necessary to enable their intended uses?**
 
+    Yes. The specification provides no more information than is needed for communication with arbitrary systems directly over TCP and UDP.
 
-## 2.1 What information might this feature expose to Web sites or other parties, and for what purposes is that exposure necessary?
+3. **How do the features in your specification deal with personal information, personally-identifiable information (PII), or information derived from them?**
 
-When the user agrees to establish a socket connection with an information system, and the connection succeeds, that system's IP address and port, and any information it sends to the user agent, becomes available to the site.
+    This specification deals with generic information.
 
-The destination system also receives the user's IP address and port, and any information the web app sends.
+    This specification allows devices to use their existing protocols for communication. These may or may not be secure. We intend to [facilitate](https://github.com/WICG/direct-sockets/blob/master/docs/explainer.md#mitigation-7) the use of TLS at some point.
 
+4. **How do the features in your specification deal with sensitive information?**
 
-## 2.2 Is this specification exposing the minimum amount of information necessary to power the feature?
+    Same as in #3.
 
-The specification provides no more information than is needed for communication with arbitrary systems directly over TCP and UDP.
+5. **Do the features in your specification introduce new state for an origin that persists across browsing sessions?**
 
-Implementors will want to carefully consider what information is revealed in the error message when connection attempts are rejected.
+    No.
 
+6. **Do the features in your specification expose information about the underlying platform to origins?**
 
-## 2.3 How does this specification deal with personal information or personally-identifiable information or information derived thereof?
+    No.
```

Comment:
For this and question 12, this API exposes local TCP/UDP port numbers to the site. Depending on how they are chosen this could leak some information about the number of active connections the system has an it creates a temporary identifier which is shared with peers.